### PR TITLE
RUBY-2673 Verify connection pools are created for all server types in direct connection mode


### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/RUBY-2673

File: spec/integration/client_construction_spec.rb [hunk 1]
```diff
@@ -100,6 +100,27 @@
...
+          opt.delete(:connect)
+          opt.update(direct_connection: true)
+        end
+      end
+
+      let(:client) do
+        new_local_client(['localhost:27019'], options)
+      end
+
+      let(:response) { client.command(ismaster: 1).documents.first }
```
Comment (comandeo) [inline spec/integration/client_construction_spec.rb:118]:
```suggestion
      let(:response) { client.command(hello: 1).documents.first }
```

**Review by comandeo — APPROVED** [review]

File: spec/integration/client_construction_spec.rb (hunk 1 above)
Comment (p-mongo) [reply spec/integration/client_construction_spec.rb:118]: This is not going to work on 3.6 is it?

Comment (comandeo) [reply spec/integration/client_construction_spec.rb:118]: Oh, my bad! Just ignore the suggestion, sorry.

**Pull request merged**